### PR TITLE
feat(api): allow passing multiple keyword arguments to `ibis.interval`

### LINKED ISSUE #5983: feat: support 2+ arguments to `ibis.interval()`
### Is your feature request related to a problem?

In Ibis, if you want to add 1 year + 1 month + 1 day to a date, you need to do:
```py
ibis.date('2023-04-12') + ibis.interval(years=1) + ibis.interval(months=1) + ibis.interval(days=1)
```
It would be nice if you could instead do:
```py
ibis.date('2023-04-12') + ibis.interval(years=1, months=1, days=1)
```
But this is not implemented and returns:
```
ValueError: Exactly one argument is required
```

### Describe the solution you'd like

Support passing multiple arguments to `ibis.interval()` and add the units together the construct the returned interval.

### What version of ibis are you running?

5.0.0

### What backend(s) are you using, if any?

_No response_

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### PR DESCRIPTION
**Pull request by jcrist**
Also does some general UX cleanups and test coverage for this utility.

Fixes #5983.

### CODE REVIEW

File: ibis/expr/api.py [hunk 1]
```diff
@@ -717,6 +717,7 @@ def _time_from_deferred(value: Deferred) -> Deferred:
 def interval(
     value: int | datetime.timedelta | None = None,
     unit: str = 's',
+    *,
```
Comment (cpcloud) [inline ibis/expr/api.py:720]: It might be good to note this as a breaking change. Up to you.

**Review by cpcloud — APPROVED** [review]
LGTM.

File: ibis/expr/api.py [hunk 2]
```diff
@@ -765,37 +766,46 @@ def interval(
...
+
+    # If no components, default to 0 s
+    if not components:
+        components.append((0, "s"))
+
+    intervals = [
+        literal(v, type=dt.Interval(u, value_type=literal(v).type()))
+        for v, u in components
+    ]
+    return functools.reduce(operator.add, intervals)
```
[the remark this answers is not in the document]
Comment (jcrist) [reply ibis/expr/api.py:808]: `sum` always adds in the `start` value (`0`), which we don't want. `functools.reduce` only includes values from the iterable, and won't call the reducer in the case of a single value (what we want).

File: ibis/expr/api.py (hunk 1 above)
Comment (jcrist) [reply ibis/expr/api.py:720]: Calling `interval(1, 's', 2)` wouldn't have worked before either. Changing the signature just makes the lack of this working more visibly explicit.

**Pull request merged**